### ISSUE TITLE
Weasyprint test fails

### ISSUE THREAD

**Issue by fabaff** [Edinburgh-Genome-Foundry/Flametree#9]
On NixOS the build process for the packages fails. I'm not sure if it's an issue with the setup or the test.

```bash
============================= test session starts ==============================
platform linux -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /build/source
collected 7 items                                                              

tests/test_basics.py ....                                                [ 57%]
tests/test_third_parties.py F..                                          [100%]

=================================== FAILURES ===================================
_______________________________ test_weasyprint ________________________________

tmpdir = local('/build/pytest-of-nixbld/pytest-0/test_weasyprint0')

    def test_weasyprint(tmpdir):
        path = str(tmpdir)
        root = file_tree(str(tmpdir))  # or 'archive.zip' to write in an archive.
        html = weasyprint.HTML(string="<b>Hello</b> world!", base_url=".")
        html.write_pdf(root._dir("pdfs")._file("test.pdf"))
        with open(os.path.join(path, "pdfs", "test.pdf"), "rb") as f:
            filesize = len(f.read())
>           assert 8000 < filesize
E           assert 8000 < 5456

tests/test_third_parties.py:26: AssertionError
=========================== short test summary info ============================
FAILED tests/test_third_parties.py::test_weasyprint - assert 8000 < 5456
========================= 1 failed, 6 passed in 1.97s ==========================
```

Thanks

**Comment by veghp** [comment]
Thank you. Admittedly, that is a very indirect test that checks the generated file's size, and perhaps some changes in weasyprint generate smaller files now. @Zulko , should we check just for presence of generated pdf file instead, or perhaps even remove this test?

**Comment by veghp** [comment]
I've [restarted the CI test](https://travis-ci.org/github/Edinburgh-Genome-Foundry/Flametree/builds/726634355) and it passes; also passes locally. Therefore this may be related to your setup -- @fabaff  can you please open and check if the pdf generated by the test is valid? It should be in `/tmp/pytest-of-username/pytest-0/test_weasyprintcurrent/pdfs/test.pdf` or similar.

**Comment by dotlambda** [comment]
This is the file generated by the test on NixOS: [test.pdf](https://github.com/Edinburgh-Genome-Foundry/Flametree/files/6565265/test.pdf)
I guess "Hello" should be bold?

**Comment by veghp** [comment]
Yes, it should be bold; on Ubuntu: [test.pdf](https://github.com/Edinburgh-Genome-Foundry/Flametree/files/6565289/test.pdf)

This suggest that weasyprint does not work as intended on your system.
(However, weasyprint is not a real dependency of flametree and is used only for the test, so there is a good argument for not making build success to be dependent on this test and third party libraries.)

**Comment by Zulko** [comment]
> This suggest that weasyprint does not work as intended on your system.

This sounds like the issue could be bumped up to the Weasy folks.

> (However, weasyprint is not a real dependency of flametree and is used only for the test, so there is a good argument for not making build success to be dependent on this test and third party libraries.)

This test (like the other tests in the same file) is actually some sort of integration test (I should probably have written that in the file). There are several EGF projects which write PDFs with the same way this test does, and the test protects Flametree against changes that would inadvertently break all these projects. So I would advise against removing it. The ideal path would be to fix Weasyprint on NixOS, or if that's too complicated adapt the test by having the last line read `assert 5000 < filesize` (instead of 8000), which is less specific but sounds reasonable to me.
